src/main/java/uk/gov/hmcts/darts/usermanagement/service/impl/SecurityGroupServiceImpl.java

```diff
@@ -82,6 +96,87 @@ public List<SecurityGroupWithIdAndRole> getSecurityGroups(List<Integer> roleIds,
             .map(securityGroupCourthouseMapper::mapToSecurityGroupWithIdAndRoleWithCourthouse).toList();
     }
 
+    @Transactional
+    @Override
+    public SecurityGroupWithIdAndRoleAndUsers modifySecurityGroup(Integer securityGroupId, SecurityGroupPatch securityGroupPatch) {
+
+        Optional<SecurityGroupEntity> securityGroupEntityOptional = securityGroupRepository.findById(securityGroupId);
+
+        if (securityGroupEntityOptional.isPresent()) {
+            SecurityGroupEntity securityGroupEntity = securityGroupEntityOptional.get();
+            updateSecurityGroupEntity(securityGroupPatch, securityGroupEntity);
+            securityGroupRepository.saveAndFlush(securityGroupEntity);
```

Comment:
I think we should assign the variable here after the saveAndFlush operation?
i.e. `securityGroupEntity = `